### PR TITLE
Change some references for ZX tutorial

### PR DESCRIPTION
**Pull request by rmoyard**
### Before submitting

Please complete the following checklist when submitting a PR:

- [ ] Ensure that your tutorial executes correctly, and conforms to the
      guidelines specified in the [README](../README.md).

- [ ] Remember to do a grammar check of the content you include.
- [ ] All tutorials conform to
      [PEP8 standards](https://www.python.org/dev/peps/pep-0008/).
      To auto format files, simply `pip install black`, and then
      run `black -l 100 path/to/file.py`.

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Title:**

**Summary:**

**Relevant references:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

**Comment by rmoyard** [comment]
@RDPEast

**Review by RDPEast — APPROVED** [review]
Looks good!

**Comment by KetpuntoG** [comment]
Hey @rmoyard 😄 Don't forget to change also the metadata references when you modify any reference on the demo

File: demonstrations/tutorial_zx_calculus.py [hunk 1]
```diff
@@ -329,10 +329,10 @@
...
 ----------------
 
-The universality of the ZX-calculus does not guarantee the existence of a simple representation, even for simple
-linear maps. For example, the Toffoli gate (the quantum AND gate) requires around 25 spiders (Z and X)! We previously
-introduced the Hadamard gate as a yellow box, which motivates the introduction of a new generator: the multi-leg H-box,
-defined as follows:
+In this section, we introduce the ZH-calculus [#Backens2018]_ .The universality of the ZX-calculus does not guarantee
+the existence of a simple representation, even for simple linear maps. For example, the Toffoli gate (the quantum AND
+gate) requires around 25 spiders (Z and X)! We previously introduced the Hadamard gate as a yellow box, which motivates
+the introduction of a new generator: the multi-leg H-box, defined as follows:
```
Comment (alvaro-at-xanadu) [inline demonstrations/tutorial_zx_calculus.py:336]: ZH calculus?

**Review by alvaro-at-xanadu — APPROVED** [review]
Looks good 👌

**Pull request merged**